### PR TITLE
feat(embodied): add evac world model

### PR DESCRIPTION
**Pull request by jzndd**
### Description
This PR integrates the EVAC World Model into the RLinf framework, providing a working end-to-end instantiation that can already run successfully.

### Motivation and Context
EVAC serves as the first fully functional example, validating the pipeline and allowing subsequent models to be dropped into the same interface.

### How has this been tested?
Successfully executed EVAC-based world model inside RLinf environment.
End-to-end rollout tested with real chunking, inference, and reward extraction.

### Additional information (optional, e.g., figures and logs):
To avoid potential compatibility issues with future base-class implementations, this PR does not remove the implementation from https://github.com/RLinf/RLinf/pull/252 . However, it updates the naming convention by renaming all occurrences of worldmodel to world_model within the corresponding directory.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 1]
```diff
@@ -0,0 +1,1213 @@
...
+        """Updates the reset state IDs for environment initialization."""
+        pass
+
+    def reset(
+        self,
+        *,
+        seed: Optional[Union[int, list[int]]] = None,
+        options: Optional[dict] = {},
+    ):
+        print("excute the reset function")
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: I have already removed this line which is just for debugging.

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 2]
```diff
@@ -0,0 +1,1213 @@
...
+        #     self.num_envs, dtype=torch.bool
+        # )
+        # self.returns = torch.zeros(
+        #     self.num_envs, dtype=torch.float32
+        # )
+
+    def _reset_metrics(self, env_idx=None):
+        if env_idx is not None:
+            # mask = torch.zeros(self.num_envs, dtype=bool, device=self.device)
+            mask = torch.zeros(self.num_envs, dtype=bool)
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fix

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 3]
```diff
@@ -0,0 +1,1213 @@
...
+    def _infer_next_chunk_rewards(
+        self,
+    ):
+        """Predict next reward using the reward model"""
+        # [num_envs, 3, 1, chunk, h, w] -> [num_envs, chunk,3, h, w]
+        num_envs, c, v, chunk, h, w = self.current_obs.shape
+        extract_chunk_obs = self.current_obs[:, :, :, -self.chunk :, :, :].permute(
+            0, 3, 1, 2, 4, 5
+        )  # [num_envs, chunk, 3, h, w]
+        # 先整理为 [num_envs * chunk, 3, h, w], 推理后获得 [num_envs * chunk, 1]
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fix

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[submodule "rlinf/models/evac/world_model/evac"]
+	path = rlinf/models/evac/world_model/evac
```
[the remark this answers is not in the document]
Comment (jzndd) [reply .gitmodules]: fix

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 5]
```diff
@@ -0,0 +1,1213 @@
...
+        )
+        episode_info["reward"] = episode_info["return"] / episode_info["episode_len"]
+        infos["episode"] = episode_info
+        return infos
+
+    def _calc_step_reward(self, chunk_rewards):
+        """Calculate step reward"""
+        reward_diffs = torch.zeros((self.num_envs, self.chunk), dtype=torch.float32).to(
+            self.device
+        )
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fix

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 6]
```diff
@@ -0,0 +1,1213 @@
...
+
+        # No terminations/truncations for now (could implement based on max steps)
+        raw_chunk_terminations = deepcopy(chunk_rewards)
+
+        raw_chunk_truncations = torch.zeros(
+            self.num_envs, self.chunk, dtype=torch.bool, device=self.device
+        )
+        # raw_chunk_truncations = torch.zeros(
+        #     num_envs, chunk_size, dtype=torch.bool
+        # )
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: removed

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 7]
```diff
@@ -0,0 +1,1213 @@
...
+
+if __name__ == "__main__":
+    from hydra import compose
+    from hydra.core.global_hydra import GlobalHydra
+    from hydra.initialize import initialize_config_dir
+
+    # Set required environment variable
+    os.environ.setdefault(
+        "EMBODIED_PATH", "/mnt/mnt/public/jzn/workspace/RLinf/examples/embodiment"
+    )
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed.

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 8]
```diff
@@ -0,0 +1,1213 @@
...
+
+    # Clear any existing Hydra instance
+    GlobalHydra.instance().clear()
+
+    config_dir = "/mnt/mnt/public/jzn/workspace/RLinf/examples/embodiment/config"
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 9]
```diff
@@ -0,0 +1,1213 @@
...
+    config_name = "libero_spatial_evac_grpo_openvlaoft_impl"
+
+    print(f"Loading config: {config_name} from {config_dir}")
+    with initialize_config_dir(config_dir=config_dir, version_base="1.1"):
+        cfg_ = compose(config_name=config_name)
+        cfg = cfg_["env"]["train"]
+    cfg.num_envs = 2
+    cfg.video_cfg.video_base_dir = "/mnt/mnt/public/jzn/workspace/latest/RLinf"
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed.

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 10]
```diff
@@ -0,0 +1,1213 @@
...
+    print("\nAfter reset:")
+    print(f"  obs keys: {list(obs.keys())}")
+
+    # Test 1: chunk_steps = self.chunk
+    print("\n" + "-" * 80)
+    chunk_steps = 8
+    num_envs = cfg.num_envs
+
+    def read_delta_actions(num_frames=8):
+        path = "/mnt/mnt/public/jzn/workspace/RLinf/reward_model/reward_data/embodiment_converted/train_data/step_0_seed_0_traj_0.npy"
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed.

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 11]
```diff
@@ -0,0 +1,1213 @@
...
+        traj = torch.clamp(traj, min=-1, max=1)
+
+        # Prepare batch
+        intrinsic_batch = (
+            self.intrinsic.unsqueeze(0).unsqueeze(0).repeat(num_envs, 1, 1, 1)
+        )
+        extrinsic_batch = i_c2w_list.repeat(num_envs, 1, 1, 1)
+
+        fps = torch.tensor([2.0]).to(dtype=torch.float32, device=self.device)
+        # fps = torch.tensor([2.0]).to(dtype=torch.float32)
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 12]
```diff
@@ -0,0 +1,1213 @@
...
+        )
+        self.returns = torch.zeros(
+            self.num_envs, device=self.device, dtype=torch.float32
+        )
+        # self.success_once = torch.zeros(
+        #     self.num_envs, dtype=torch.bool
+        # )
+        # self.returns = torch.zeros(
+        #     self.num_envs, dtype=torch.float32
+        # )
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed.

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 13]
```diff
@@ -0,0 +1,1213 @@
...
+            #     (
+            #         self.all_samples[:, :, idx_history],  # [num_envs, c, n_previous-1, h, w]
+            #         self.all_samples[:, :, -1:]
+            #         .repeat(1, 1, self.chunk + 1, 1, 1),  # [num_envs, c, chunk+1, h, w]
+            #     ),
+            #     dim=2,
+            # ).to(
+            #     dtype=self.inference_dtype
+            # )  # [num_envs, c, n_previous+chunk, h, w]
+
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed.

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 14]
```diff
@@ -0,0 +1,1213 @@
...
+            )  # [3, n_previous, H, W]
+
+            img_tensors.append(img_tensor)
+
+        self.current_obs = []
+        for img_tensor in img_tensors:
+            # [3, n_previous, H, W] -> [1, 3, 1, n_previous, H, W]
+            env_obs = img_tensor.unsqueeze(1)
+            self.current_obs.append(env_obs)
+        # self.current_obs = torch.stack(self.current_obs, dim=0) # [num_envs, 3, 1, n_previous, H, W]
```
[the remark this answers is not in the document]
Comment (jzndd) [reply rlinf/envs/world_model/world_model_evac_env.py]: fixed.

File: .gitmodules [hunk 15]
```diff
@@ -0,0 +1,3 @@
+[submodule "rlinf/models/world_model/evac"]
+	path = rlinf/models/evac/world_model/evac
+	url = https://github.com/jzndd/EnerVerse-AC.git
```
Comment (liyd) [inline .gitmodules]: 建议把repo移到rlinf组织下，权限相关的内容麻烦 @guozhen1997 看下

File: examples/embodiment/config/env/train/libero_spatial_evac.yaml [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+
+model_type: evac
+gen_num_image_each_step: 8
+
+chunk: 8
+n_previous: 4
+n_view: 1
+
+sample_size: [256, 256]
+initial_image_path: /mnt/mnt/public/jzn/workspace/RLinf-backup/reward_model/world_model_buffer/
```
Comment (liyd) [inline examples/embodiment/config/env/train/libero_spatial_evac.yaml]: 路径相关的配置需要都改成默认待配置的状态

File: rlinf/envs/world_model/base_world_env.py [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Optional, Union
+
+import torch
+
+
+class BaseWorldEnv(ABC):
```
Comment (guozhen1997) [inline rlinf/envs/world_model/base_world_env.py:25]: please add `chunk_step` and `reset` as abstract method

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 18]
```diff
@@ -0,0 +1,1165 @@
...
+
+        infos["final_observation"] = final_obs
+        infos["final_info"] = final_info
+        infos["_final_info"] = dones
+        infos["_final_observation"] = dones
+        infos["_elapsed_steps"] = dones
+
+        return extracted_obs, infos
+
+    def chunk_step(self, policy_output_action):
```
Comment (liyd) [inline rlinf/envs/world_model/world_model_evac_env.py]: 这里有个疑问，如果policy model输出的chunk steps和world model的chunk steps应该会有问题？正常的simulator因为是单步渲染，不存在这个问题

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 19]
```diff
@@ -0,0 +1,1165 @@
...
+
+        return obs
+
+    def policy_output_to_abs_action(self, policy_output):
+        """Convert policy output to absolute action"""
+        # policy output: [num_envs, chunk, 7]
+        # abs_action_mean: [1, 8]
+        # abs_action_std: [1, 8]
+        policy_output = torch.tensor(policy_output, device=self.device).float()
+        pre_ee_pose = self.action_buffer[:, -self.chunk - 1, :8]
```
Comment (liyd) [inline rlinf/envs/world_model/world_model_evac_env.py]: 8是不是设为一个超参更合适？

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 20]
```diff
@@ -0,0 +1,1165 @@
...
+        for env_idx in range(num_envs):
+            action_np = action_buffer[env_idx].detach().cpu().numpy()
+            _, delta_action = get_action_from_abs_act(action_np)
+            if isinstance(delta_action, np.ndarray) and len(delta_action.shape) > 1:
+                delta_action = delta_action[-1]  # Take only the last one
+            delta_actions.append(delta_action)
+
+        return torch.from_numpy(np.stack(delta_actions, axis=0))  # [num_envs, 16]
+
+    def _handle_auto_reset(self, dones, extracted_obs, infos):
```
Comment (liyd) [inline rlinf/envs/world_model/world_model_evac_env.py]: 这里逻辑是说任意一条轨迹被完成，就重置所有的轨迹吗？

File: examples/embodiment/config/libero_spatial_evac_grpo_openvlaoft.yaml [hunk 21]
```diff
@@ -0,0 +1,203 @@
...
+    dir: .
+  output_subdir: null
+  searchpath:
+    - file://${oc.env:EMBODIED_PATH}/config/
+
+cluster:
+  num_nodes: 1
+  component_placement:
+    actor,rollout,env: 0,1,2,3,4,5,6,7
+    # actor: 4,5,6,7
```
Comment (guozhen1997) [inline examples/embodiment/config/libero_spatial_evac_grpo_openvlaoft.yaml]: remove L17-19

File: rlinf/envs/world_model/world_model_evac_env.py [hunk 22]
```diff
@@ -0,0 +1,1170 @@
...
+        episode_info["reward"] = episode_info["return"] / episode_info["episode_len"]
+        infos["episode"] = episode_info
+        return infos
+
+    def _calc_step_reward(self, chunk_rewards):
+        """Calculate step reward"""
+        reward_diffs = torch.zeros(
+            (self.num_envs, self.chunk), dtype=torch.float32, device=self.device
+        )
+        # reward_diffs = torch.zeros((self.num_envs, self.chunk), dtype=torch.float32)
```
Comment (guozhen1997) [inline rlinf/envs/world_model/world_model_evac_env.py]: remove this line

**Review by XuS1994 — APPROVED** [review]
